### PR TITLE
Platformize core configuration subprojects

### PR DESCRIPTION
**Pull request by alllex**
The following projects are moved to the `core-configuration` platform in this PR:
- api-metadata
- file-collections
- model-core
- model-groovy
- base-services-groovy

### Reviewing cheatsheet

Before merging the PR, comments starting with 
- ❌ ❓**must** be fixed
- 🤔 💅 **should** be fixed
- 💭 **may** be fixed
- 🎉 celebrate happy things

### CODE REVIEW

**Comment by alllex** [comment]
@bot-gradle test this

File: settings.gradle.kts [hunk 1]
```diff
@@ -119,13 +114,18 @@ platform("core-runtime") {
 
 // Core Configuration Platform
 platform("core-configuration") {
+    subproject("api-metadata")
```
Comment (bamboo) [inline settings.gradle.kts:117]: 👍 kudos for keeping it sorted

**Review by bamboo — APPROVED** [review]
💯

**Comment by alllex** [comment]
@bot-gradle test and merge

**Review by tresat — APPROVED** [review]

**Pull request merged**